src/benchmark.cc

```diff
@@ -845,6 +845,13 @@ void MaybeReenterWithoutASLR(int /*argc*/, char** argv) {
   // Have we failed to change the personality? That may happen.
   if (prev_personality == -1) return;
 
+  // Make sure the parsona has been updated,
+  // otherwise we will try to reenter infinitely.
+  // This seems impossible, but can happen in some docker configurations.
+  const auto new_personality = personality(0xffffffff);
+  if (internal::get_as_unsigned(new_personality) != proposed_personality)
```

Comment:
Hm, i would think we should just check that `ADDR_NO_RANDOMIZE` flag is now set.
Mainly, i don't think it is _guaranteed_ that setting one flag wouldn't set/unset something else.

Reply:
Changed.